django/db/models/sql/query.py

```diff
@@ -1192,8 +1192,8 @@ def build_lookup(self, lookups, lhs, rhs):
         # stage because join promotion can't be done in the compiler. Using
         # DEFAULT_DB_ALIAS isn't nice but it's the best that can be done here.
         # A similar thing is done in is_nullable(), too.
-        if (connections[DEFAULT_DB_ALIAS].features.interprets_empty_strings_as_nulls and
-                lookup_name == 'exact' and lookup.rhs == ''):
+        if (lookup_name == 'exact' and lookup.rhs == '' and
+                connections[DEFAULT_DB_ALIAS].features.interprets_empty_strings_as_nulls):
```

Comment:
As you're reordering here anyway, I'd reformat this to:
```suggestion
        if (
            lookup_name == 'exact' and lookup.rhs == '' and
            connections[DEFAULT_DB_ALIAS].features.interprets_empty_strings_as_nulls
        ):
```